_posts/2020-09-22-data-explorer-kafka-connect.md

```diff
@@ -0,0 +1,440 @@
+
+## Base installation
+
+Start by installing the Strimzi Operator and use it to spin up a single-node Kafka Cluster on Kubernetes. Installing Strimzi using `Helm` is pretty easy:
+
+```shell
+helm repo add strimzi https://strimzi.io/charts/
+helm install strimzi-kafka strimzi/strimzi-kafka-operator
+```
+
+To confirm successful installation:
+
+```
+kubectl get pods -l=name=strimzi-cluster-operator
+```
+
+You should see the cluster operator `Pod` in `Running` status
+
+```shell
+NAME                                        READY   STATUS    RESTARTS   AGE
+strimzi-cluster-operator-5c66f679d5-69rgk   1/1     Running   0          43s
+```
+
+To deploy a single-node kafka cluster (along with Zookeeper):
+
+```shell
+kubectl apply -f https://github.com/abhirockzz/kusto-kafka-connect-strimzi/raw/master/deploy/kafka.yaml
+```
+
+Wait for the cluster to start:
+
+```shell
+kubectl get pod my-kafka-cluster-kafka-0 -w
+```
+
+The Kafka Pod should transition to `Running` status and both the containers should be in `READY` state
+
+```shell
+NAME                       READY   STATUS    RESTARTS   AGE
+my-kafka-cluster-kafka-0   2/2     Running   0          1m
+```
+
+## Kafka Connect cluster setup
+
+The Strimzi container images for Kafka Connect include two built-in file connectors - `FileStreamSourceConnector` and `FileStreamSinkConnector`. For the purposes of this blog, a custom Docker image seeded with [Azure Data Explorer connector](https://github.com/Azure/kafka-sink-azure-kusto/releases/tag/v1.0.1) (version `1.0.1`) is [available on Docker Hub](https://hub.docker.com/r/abhirockzz/adx-connector-strimzi) and it is referenced in the `KafkaConnect` resource definition (`image: abhirockzz/adx-connector-strimzi:1.0.1`):
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  image: abhirockzz/adx-connector-strimzi:1.0.1
+  version: 2.4.0
+....
+```
+
+If you want to build your own Docker image, use the [Strimzi Kafka Docker image](https://hub.docker.com/r/strimzi/kafka) as a base and add the Azure Data Explorer connector JAR top to the plugin path. Start by downloading the connector JAR file:
+
+```shell
+export KUSTO_KAFKA_SINK_VERSION=1.0.1
+mkdir connector && cd connector
+curl -L -O https://github.com/Azure/kafka-sink-azure-kusto/releases/download/v$KUSTO_KAFKA_SINK_VERSION/kafka-sink-azure-kusto-$KUSTO_KAFKA_SINK_VERSION-jar-with-dependencies.jar
+```
+
+Then, you can use this `Dockerfile` to build the Docker image:
+
+```dockerfile
+FROM strimzi/kafka:latest-kafka-2.4.0
```

Comment:
Since _latest_ is basically our master, it is sometimes dynamically changing. Can you change this to `FROM strimzi/kafka:0.19.0-kafka-2.4.0` so that people who copy this base on the stable release?

Reply:
updated. thank you pointing this out!